### PR TITLE
cli: support NEP11-related commands

### LINKED ISSUE #1860: CLI support for NEP-11 and NNS
Registration/administration/transfers.

### LINKED ISSUE #1864: rpc: support all NEP11 methods which return iterators
See https://github.com/nspcc-dev/neo-go/pull/1857#discussion_r601805223.

Methods list: 
* Tokens (optional method)
* OwnerOf (divisible method)
* TokensOf? (common method)

### PR DESCRIPTION
**Pull request by AnnaShaleva**
Close #1860, close #1864.

TODO: 

- [x] tests for `balance` and `transfer` (something is wrong with transfer implementation or NFT example, investigating)
- [x] formatting

### CODE REVIEW

File: cli/wallet/nep17.go [hunk 1]
```diff
@@ -64,6 +64,7 @@ var (
 )
 
 func newNEP17Commands() []cli.Command {
+	balanceFlags := append(baseBalanceFlags, options.RPC...)
```
Comment (roman-khimov) [inline cli/wallet/nep17.go:63]: You need to copy here, otherwise this append can clash with NEP11 append.

File: cli/wallet/nep11.go [hunk 2]
```diff
@@ -1,12 +1,34 @@
...
+	"github.com/nspcc-dev/neo-go/pkg/wallet"
 	"github.com/urfave/cli"
 )
 
 func newNEP11Commands() []cli.Command {
+	tokenID := cli.StringFlag{
+		Name:  "id",
+		Usage: "Token ID",
+	}
+	balanceFlags := append(baseBalanceFlags, tokenID)
```
Comment (roman-khimov) [inline cli/wallet/nep11.go:26]: Copy.

File: cli/wallet/nep11.go [hunk 3]
```diff
@@ -1,12 +1,34 @@
...
+	balanceFlags = append(baseBalanceFlags, options.RPC...)
```
Comment (roman-khimov) [inline cli/wallet/nep11.go:27]: `append(balanceFlags...`? Otherwise you lose `tokenID`.

File: cli/wallet/nep17.go [hunk 4]
```diff
@@ -43,18 +43,14 @@ var (
...
 		outFlag,
 		fromAddrFlag,
 		toAddrFlag,
 		tokenFlag,
 		gasFlag,
-		cli.StringFlag{
-			Name:  "amount",
-			Usage: "Amount of asset to send",
-		},
-	}, options.RPC...)
```
Comment (roman-khimov) [inline cli/wallet/nep17.go]: Amount is still applicable for divisible NFTs.

File: cli/wallet/nep17.go [hunk 5]
```diff
@@ -65,6 +61,12 @@ var (
 
 func newNEP17Commands() []cli.Command {
 	balanceFlags := append(baseBalanceFlags, options.RPC...)
+	transferFlags := append(baseTransferFlags,
```
Comment (roman-khimov) [inline cli/wallet/nep17.go:64]: Copy.

**Review by roman-khimov — COMMENTED** [review]
It'd be nice to test `properties` of course.

File: pkg/rpc/client/nep11.go [hunk 6]
```diff
@@ -123,7 +123,7 @@ func (c *Client) TransferNEP11D(acc *wallet.Account, to util.Uint160,
 	if err != nil {
 		return util.Uint256{}, fmt.Errorf("bad account address: %w", err)
 	}
-	tx, err := c.createNEP11TransferTx(acc, tokenHash, gas, cosigners, acc.Address, from, to, amount, tokenID)
+	tx, err := c.createNEP11TransferTx(acc, tokenHash, gas, cosigners, from, to, amount, tokenID)
```
Comment (AnnaShaleva) [inline pkg/rpc/client/nep11.go]: Tests are TBD in https://github.com/nspcc-dev/neo-go/issues/1940.

File: pkg/rpc/response/result/invoke.go [hunk 7]
```diff
@@ -76,6 +108,29 @@ func (r *Invoke) UnmarshalJSON(data []byte) error {
...
+						iteratorValues[j], err = stackitem.FromJSONWithTypes(iteratorAux.Value[j])
+						if err != nil {
+							err = fmt.Errorf("failed to unmarshal iterator values: %w", err)
+							break
+						}
+					}
+					// it's impossible to restore initial iterator type; also iterator is almost
+					// useless outside of the VM, thus let's replace it with the most general
+					// iterable type (Array)
+					// TODO: @roman-khimov, option #2 is to create iterator over array:
```
Comment (AnnaShaleva) [inline pkg/rpc/response/result/invoke.go:125]: We need to decide which one would be better to use. At the current moment, option#1 is implemented.

File: pkg/smartcontract/manifest/manifest.go [hunk 8]
```diff
@@ -244,3 +247,11 @@ func (m *Manifest) FromStackItem(item stackitem.Item) error {
 	m.Extra = extra
 	return nil
 }
+
+// IsValid returns whether the provided string is a valid NEP-token standard name.
+func IsValid(standard NEPStandardName) bool {
+	st := string(standard)
+	return st == NEP10StandardName ||
```
[the remark this answers is not in the document]
Comment (AnnaShaleva) [reply pkg/smartcontract/manifest/manifest.go]: It's a remnant, I removed it.

File: pkg/rpc/response/result/invoke.go (hunk 7 above)
Comment (roman-khimov) [reply pkg/rpc/response/result/invoke.go:125]: Option 2 won't work after #1935. And we need to keep `truncated` flag somehow. So maybe we need to store some special `struct` there with `truncated` and `values`? Anyone could try the type of the `Value()` then and see if it's the thing he expects.

File: pkg/smartcontract/manifest/manifest.go [hunk 9]
```diff
@@ -24,6 +24,9 @@ const (
 	NEP17Payable = "NEP-17-Payable"
 )
 
+// NEPStandardName represents supported token standard name.
+type NEPStandardName string
```
[the remark this answers is not in the document]
Comment (AnnaShaleva) [reply pkg/smartcontract/manifest/manifest.go:28]: I think that it's more convenient and developer-friendly to have a separate type for these two standards for common NEP methods. I'll fix the type of untyped constants.

File: pkg/rpc/response/result/invoke.go (hunk 7 above)
Comment (AnnaShaleva) [reply pkg/rpc/response/result/invoke.go]: Nice solution, implemented.

File: pkg/smartcontract/manifest/manifest.go (hunk 9 above)
[the remark this answers is not in the document]
Comment (AnnaShaleva) [reply pkg/smartcontract/manifest/manifest.go:28]: Fixed.

[the remark this answers is not in the document]
Comment (roman-khimov) [reply pkg/smartcontract/manifest/manifest.go:28]: Just looking at the number of casts added I don't think it's a good idea. Maybe we don't really need `NEPStandardName`?

**Review by roman-khimov — APPROVED** [review]

**Pull request merged**